presto-cache/src/main/java/com/facebook/presto/cache/filemerge/FileMergeCacheManager.java

```diff
@@ -161,17 +166,43 @@ public boolean get(FileReadRequest request, byte[] buffer, int offset)
             stats.incrementCacheMiss();
         }
 
-        return result;
+        if (!result && hiveFileContext.getQuota().isPresent() && getCacheSize(hiveFileContext.getTable()) + request.getLength() > hiveFileContext.getQuota().getAsLong()) {
```

Comment:
+1

Reply:
I dont think I understand this logic. Why get method has anything to do with cache quota?
Get method should just return the data if the data is present in the cache?
It is the put method responsibility to put the data in the cache or not. While putting more data to the cache, it should be checked if the new data can be accommodated in the cache or not.

Reply:
Same, we don't wanna check quota on read. Move the checking and persisting cache logic to write.

Reply:
Originally we 
``
// hint the cache no matter what
cache.get(request.getPath(), () -> true);
``
because we need to update the cache so that it can be marked as recently accessed.
but if it exceeds the cache quota, we can’t put it in cache any more

and we rely on read result to decide if we want to read and put to cache, that’s why we have those logic in read path